.circleci/config.yml

```diff
@@ -1,64 +1,142 @@
           keys:
-            - javascript-cache-v1-{{ checksum "package.json" }}
-
+            - javascript-dependencies-v1-{{ checksum "yarn.lock" }}
       - run:
+          name: Install Yarn
           command: yarn install --ignore-engines
-
       - save_cache:
-          key: javascript-cache-v1-{{ checksum "package.json" }}
           paths:
             - node_modules
+          key: javascript-dependencies-v1-{{ checksum "yarn.lock" }}
 
-      # setup db
+  run_rubocop:
+    docker:
+      - image: circleci/ruby:2.5.1-node-browsers
+    working_directory: ~/spark-starter-kit
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - ruby-dependencies-v6-{{ checksum "Gemfile.lock" }}
+      - restore_cache:
+          keys:
+            - javascript-dependencies-v1-{{ checksum "yarn.lock" }}
       - run:
-          name: Wait for DB
-          command: dockerize -wait tcp://localhost:5432 -timeout 1m
+          name: Set bundle path
+          command: bundle config --local path vendor/bundle
+      - run:
+          name: Rubocop Check
+          command: bundle exec rubocop || true
 
+  run_reek:
+    docker:
+      - image: circleci/ruby:2.5.1-node-browsers
+    working_directory: ~/spark-starter-kit
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - ruby-dependencies-v6-{{ checksum "Gemfile.lock" }}
+      - restore_cache:
+          keys:
+            - javascript-dependencies-v1-{{ checksum "yarn.lock" }}
+      - run:
+          name: Set bundle path
+          command: bundle config --local path vendor/bundle
+      - run:
+          name: Reek Check
+          command: bundle exec reek || true
+
+  run_brakeman:
+    docker:
+      - image: circleci/ruby:2.5.1-node-browsers
+    working_directory: ~/spark-starter-kit
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - ruby-dependencies-v6-{{ checksum "Gemfile.lock" }}
+      - restore_cache:
+          keys:
+            - javascript-dependencies-v1-{{ checksum "yarn.lock" }}
+      - run:
+          name: Set bundle path
+          command: bundle config --local path vendor/bundle
       - run:
-          name: Setup databse
-          command: bundle exec rake db:schema:load --trace
+          name: Brakeman Check
+          command: bundle exec brakeman --skip-files vendor/bundle/
 
-      # rspec
+  run_tests:
+    docker:
+      - image: circleci/ruby:2.5.1-node-browsers
+      - image: postgres:10.4
+    working_directory: ~/spark-starter-kit
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - ruby-dependencies-v6-{{ checksum "Gemfile.lock" }}
+      - restore_cache:
+          keys:
+            - javascript-dependencies-v1-{{ checksum "yarn.lock" }}
+      - run:
+          name: Set bundle path
+          command: bundle config --local path vendor/bundle
       - run:
-          name: Run rspec
+          name: Wait for DB
```

Comment:
do we need that?

Reply:
yeah, I think it might be useful to wait for PG port, just in case.